bee-protocol/src/workers/index_updater.rs

```diff
@@ -116,17 +114,14 @@ async fn update_past_cone<B: StorageBackend>(
             continue;
         }
 
-        tangle
-            .update_metadata(&parent_id, |metadata| {
-                metadata.set_milestone_index(index);
-                // TODO: That was fine in a synchronous scenario, where this algo had the newest information, but
-                // probably isn't the case in the now asynchronous scenario. Investigate!
-                metadata.set_omrsi(IndexId::new(index, parent_id));
-                metadata.set_ymrsi(IndexId::new(index, parent_id));
-            })
-            .await;
-
-        if let Some(parent) = tangle.get(&parent_id).await {
+        tangle.update_metadata(&parent_id, |metadata| {
+            metadata.set_milestone_index(index);
```

Comment:
Just as a reminder that we might want to add it here as well: PR #1255 adds a trace log here to allow monitoring this `update_metadata` call.

Reply:
Yeah there is probably way more missing than just trace logs tbh, we'll resolve that when we do the final production/shimmer merge 👍